src/Client.cpp

```
@@ -2,43 +2,188 @@
-	this->_username = c._username;
-	this->_channels = c._channels;
-}
+Client::Client(const Client &c) :	_fd(c._fd),
+																	_address(c._address),
+																	_hostname(c._hostname),
+																	_nickname(c._nickname),
+																	_username(c._username),
+																	_realname(c._realname),
+																	_channels(c._channels),
+																	_status(c._status),
+																	_mode(c._mode)
+{}
 
 Client&
 Client::operator=(const Client &c) {
-	this->_nick = c._nick;
-	this->_address = c._address;
-	this->_username = c._username;
+	this->_fd = c.getFd();
+	this->_address = c.getAddress();
+	this->_hostname = c.getHostname();
+	this->_nickname = c.getNickname();
+	this->_username = c.getUsername();
+	this->_realname = c.getRealname();
 	this->_channels = c._channels;
+	this->_status = c.getStatus();
+	this->_mode = c._mode;
 	return *this;
 }
 
 Client::~Client() {
 	// TODO: Check and exit from channels
-	LOG_INFO("Removed client: " << this->_nick)
+	close(this->_fd);
+	LOG_INFO("Removed client: " << this->_nickname)
+}
+
+int
+Client::getFd() const {
+	return this->_fd;
+}
+
+std::string
+Client::getAddress() const {
+	return this->_address;
+}
+
+std::string
+Client::getHostname() const {
+	return this->_hostname;
+}
+
+std::string
+Client::getNickname() const {
+	return this->_nickname;
+}
+
+std::string
+Client::getUsername() const {
+	return this->_username;
+}
+
+std::string
+Client::getRealname() const {
+	return this->_realname;
+}
+
+ClientStatus
+Client::getStatus() const {
+	return this->_status;
+}
+
+// std::vector<Channel&>
+// Client::getChannels() const {
+// 	TODO
+// }
+
+bool
+Client::getMode(char c) const
+{
+	switch (c)
+	{
+		case 'a' : return this->_mode.a;
+		case 'i' : return this->_mode.i;
+		case 'w' : return this->_mode.w;
+		case 'r' : return this->_mode.r;
+		case 'o' : return this->_mode.o;
+		case 'O' : return this->_mode.O;
+		case 's' : return this->_mode.s;
+		default : return false;
```

Functions shouldn't silently fail. Throwing an exception would be a better alternative